src/Interpreters/HashJoin.cpp

```diff
@@ -744,7 +744,7 @@ bool HashJoin::addJoinedBlock(const Block & source_block, bool check_limits)
     size_t total_rows = 0;
     size_t total_bytes = 0;
     {
-        if (storage_join_lock.mutex())
+        if (storage_join_lock)
             throw DB::Exception("addJoinedBlock called when HashJoin locked to prevent updates",
                                 ErrorCodes::LOGICAL_ERROR);
```

Comment:
It's not changed in this PR, but I wonder why it's `LOGICAL_ERROR`, can it be reached in we perform `INSERT` into `JOIN Engine` table during `SELECT`?

_UPD:_ because lock already acquired in `StorageJoin::insertBlock`, but `storage_join_lock` can be set only from `StorageJoin::getJoinLocked` that are used for `SELECT`ing data.